src/fs/file.rs

```diff
@@ -377,6 +377,41 @@ impl<'dir> File<'dir> {
                 .is_some_and(|p| all_mounts().contains_key(p))
     }
 
+    /// Whether the directory represents a Btrfs subvolume
+    #[cfg(target_os = "linux")]
+    pub fn is_btrfs_subvolume(&self) -> bool {
+        // Listing all subvolumes with ioctl(BTRFS_IOC_TREE_SEARCH) requires CAP_SYS_ADMIN, normal users can't do that.
+        // So we test the inode number, directory representing a subvolume has always inode number 256
+        self.metadata.file_type().is_dir()
+            && self.metadata.ino() == 256
+            && self.is_btrfs().unwrap_or(false)
+    }
+
+    #[cfg(not(target_os = "linux"))]
+    pub fn is_btrfs_subvolume(&self) -> bool {
+        false
+    }
+
+    #[cfg(target_os = "linux")]
+    fn is_btrfs(&self) -> io::Result<bool> {
+        use std::os::unix::ffi::OsStrExt;
+        for part in self.absolute_path().unwrap_or(&self.path).ancestors() {
+            if let Some(mount) = all_mounts().get(part) {
+                return Ok(mount.fstype == "btrfs");
```

Comment:
nit: can we make this a constant?

Reply:
You mean adding a `"btrfs"` constant?
```diff
+ const BTRFS_FSTYPE_NAME: &str = "btrfs";
```

Reply:
:+1: yeah, everything looks good tho, looks like the tests are resolved. Appreciate the contribution :heart: